Core/Src/drivers/nand_flash/nand_m79a_lld.c

```diff
@@ -395,17 +395,17 @@ NAND_ReturnType NAND_Page_Program(PhysicalAddrs *addr, uint16_t length, uint8_t
     SPI_Params exec_cmd = {.buffer = command_exec, .length = 4};
 
     if (NAND_SPI_Send(&exec_cmd) != SPI_OK) {
-        return Ret_WriteFailed;
+        return Ret_Failed;
```

Comment:
This seems less distinct? What am I missing?

Reply:
Ret_WriteFailed indicates the block failed to program so it may be a bad block. Just returning failure is nonspecific, but in this case it would be an SPI failure

Reply:
I'm confused - it looks like you are returning a nonspecific failure. Why aren't we returning Ret_SPI_Failure? And could we even differentiate between different SPI failures? (We probably only have read timeout.) And we should be logging all failures explicitly so we can debug this from the ground.

Reply:
That wasn't one of the available return types. Should I add it?

Reply:
If there are only two possible ways to fail (e.g. NAND timeout and SPI timeout) then Ret_Failed is the same as Ret\_Bus\_Err (seems like Ret_SPIFailed is a little *too* specific). This assumes that NAND\_SPI\_Send is effectively a boolean: SPI\_OK or SPI\_Failed, otherwise we should try to pass the specific SPI error up.